src/main/java/io/vertx/core/http/impl/ClientConnection.java

```diff
@@ -242,8 +248,12 @@ private void handshakeComplete(ChannelHandlerContext ctx, FullHttpResponse respo
       }
       ws = new WebSocketImpl(vertx, ClientConnection.this, supportsContinuation, client.getOptions().getMaxWebsocketFrameSize());
       handshaker.finishHandshake(channel, response);
+
       log.debug("WebSocket handshake complete");
-      wsConnect.handle(ws);
+      context.executeFromIO(() -> {
```

Comment:
when I execute the test (WebsocketTest#testClientWorker), the channel thread is an eventloop thread, not sure to understand what you mean.